modules/document_repository/php/edit.class.inc

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * This file contains the NDB_Form_document_repository class
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Document_Repository
+ * @author   Shen Wang <wangshen.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS\document_repository;
+
+/**
+ * Creates the form for document_repository/edit page
+ *
+ * @category Loris
+ * @package  Document_Repository
+ * @author   Shen Wang <wangshen.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Edit extends \NDB_Form
+{
+    public $skipTemplate = true;
+
+    /**
+     * Entry point for /document_repository/edit/
+     * Checks if file id is provided and otherwise redirects to
+     * /document_repository/ page
+     *
+     * @return void
+     */
+    function setup()
+    {
+        parent::setup();
+
+        $db      = \Database::singleton();
+        $factory = \NDB_Factory::singleton();
+        $baseURL = $factory->settings()->getBaseURL();
+
+        $idDocFile = $_GET['id'];
```

Comment:
The id should be extracted from the PSR7 request object in the handle function and put into a class variable if required, not come from super-globals.

(In fact, I don't see why this setup function exists at all, it looks like it should pretty much all be in handle..)

I also think it would make more sense for the id to be part of the URL, not part of a query param. ie. `/document_repository/edit/1` rather than `/document_repository/edit?id=1`

Reply:
fixed